src/commands/delete.js

```diff
@@ -0,0 +1,48 @@
+const utils = require('../utils');
+
+const _delete = (context, version) => {
+  if (!version) {
+    context.line('Error: No version - provide a version like "1.0.0"...\n');
+    return Promise.resolve(true);
+  }
+  return utils.checkCredentials()
+    .then(() => utils.getLinkedApp())
+    .then((app) => {
+      context.line(`Preparing to delete version ${version} of your app "${app.title}".\n`);
+      const url = `/apps/${app.id}/versions/${version}`;
+      utils.printStarting(`Deleting ${version}`);
+      return utils.callAPI(url, {
+        method: 'DELETE',
+      });
+    })
+    .then(() => {
+      utils.printDone();
+      context.line('  Deletion successful!\n');
+    });
+};
+_delete.argsSpec = [
+  {name: 'version', example: '1.0.0', required: true, help: 'the version to delete'},
+];
+_delete.argOptsSpec = {};
+_delete.help = 'Delete a version of your app as long as it has no users/Zaps.';
+_delete.example = 'zapier delete 1.0.0';
+_delete.docs = `\
+A utility to allow deleting app versions that aren't used.
+
+> The app version needs to have no users/Zaps in order to be deleted.
+
+**Arguments**
+
+${utils.argsFragment(_delete.argsSpec)}
+${utils.argOptsFragment(_delete.argOptsSpec)}
+
+${'```'}bash
+$ zapier delete 1.0.0
```

Comment:
`delete` from a code perspective is solid, however, we should discuss the UX.

First up is whether we should be consistent within the CLI tool. For `invite` and `env`, we've chosen `--remove` as the way to nuke a specific piece of data. This introduces a different pattern. Granted, in those cases, there is a required argument, so the commands enforce what is getting removed. With `versions`, we'd have to allow an optional argument to specify which one. Still, I think that may be preferable to `zapier delete`.

A couple other considerations:

1) From the command alone, it's not clear what `delete` is going to allow you to remove. Apps? App Versions?
 Both? More stuff? Also, when we decide to allow deleting apps, what are we going to pick as the command if delete is already reserved for versions?
2) Related to one, should we go ahead and allow deleting apps as well? Right now you can nuke the versions, but then you'd have to go into the UI to nuke the app itself, which is a little inconsistent. I've heard devs ask in the public Slack channel to delete an test app they used to experiment, so there is a use case for deleting an entire app in one go.